### PR TITLE
Add da service to nomos node

### PR DESCRIPTION
**Pull request by Zeegomo**
This PR:
* updates the data availability service to work with generic da protocols
* add the data availability service to the nomos node

Could actually split this but it's not too big and you can find the features grouped by commits anyway

### CODE REVIEW

File: nomos-core/src/da/mod.rs [hunk 1]
```diff
@@ -13,7 +13,10 @@ pub trait DaProtocol {
     type Blob: Blob;
     type Attestation: Attestation;
     type Certificate: Certificate;
+    type Settings: Clone;
 
+    // Construct a new instance
+    fn new(settings: Self::Settings) -> Self;
```
Comment (danielSanchezQ) [inline nomos-core/src/da/mod.rs:19]: yep, I always forget about this.

**Comment by danielSanchezQ** [comment]
I rebased it to solve some conflicts, but now it just do not work for waku. I think it was working before. Can you confirm what did I miss @Zeegomo. Sorry!

**Review by danielSanchezQ — APPROVED** [review]

**Review by youngjoon-lee — APPROVED** [review]
👍

**Review by al8n — APPROVED** [review]

**Comment by Zeegomo** [comment]
> I rebased it to solve some conflicts, but now it just do not work for waku. I think it was working before. Can you confirm what did I miss @Zeegomo. Sorry!

I guess some conditional compilation stuff as I only implemented it for the libp2p backend

**Comment by danielSanchezQ** [comment]
> > I rebased it to solve some conflicts, but now it just do not work for waku. I think it was working before. Can you confirm what did I miss @Zeegomo. Sorry!
> 
> I guess some conditional compilation stuff as I only implemented it for the libp2p backend, let me check

I think at this point we can ditch waku in the nomos binary itself (where the problems are iirc).

**Comment by Zeegomo** [comment]
For now it's just a fix of a few lines, if we find it's increasingly difficult we can remove it in a different PR

**Pull request merged**